src/Microsoft.Tye.Core/DockerContainerBuilder.cs

```diff
@@ -34,55 +34,93 @@ public static async Task BuildContainerImageAsync(OutputContext output, Applicat
                 throw new ArgumentNullException(nameof(container));
             }
 
-            using var tempFile = TempFile.Create();
-
+            string contextDirectory;
             var dockerFilePath = Path.Combine(project.ProjectFile.DirectoryName, "Dockerfile");
-            if (File.Exists(dockerFilePath))
-            {
-                output.WriteDebugLine($"Using existing Dockerfile '{dockerFilePath}'.");
-            }
-            else
-            {
-                await DockerfileGenerator.WriteDockerfileAsync(output, application, project, container, tempFile.FilePath);
-                dockerFilePath = tempFile.FilePath;
-            }
 
-            // We need to know if this is a single-phase or multi-phase Dockerfile because the context directory will be
-            // different depending on that choice.
-            string contextDirectory;
-            if (container.UseMultiphaseDockerfile ?? true)
-            {
-                contextDirectory = ".";
-            }
-            else
+            TempFile? tempFile = null;
+            TempDirectory? tempDirectory = null;
+
+            try
             {
-                var publishOutput = project.Outputs.OfType<ProjectPublishOutput>().FirstOrDefault();
-                if (publishOutput is null)
+                // We need to know if this is a single-phase or multi-phase Dockerfile because the context directory will be
+                // different depending on that choice.
+                //
+                // For the cases where generate a Dockerfile, we have the constraint that we need
+                // to place it on the same drive (Windows) as the docker context.
+                if (container.UseMultiphaseDockerfile ?? true)
+                {
+                    // For a multi-phase Docker build, the context is always the project directory.
+                    contextDirectory = ".";
+
+                    if (File.Exists(dockerFilePath))
+                    {
+                        output.WriteDebugLine($"Using existing Dockerfile '{dockerFilePath}'.");
+                    }
+                    else
+                    {
+                        // We need to write the file, let's stick it under obj.
+                        Directory.CreateDirectory(Path.Combine(project.ProjectFile.DirectoryName, "obj"));
```

Comment:
Instead of hard coding obj use the variable from msbuild `IntermediateOutputPath` (I think that's what it's called).

Reply:
Adding it. FYI this code path doesn't have a UX right now, so you have no way to hit it. I'm just maintaining the code because we'll want to expose it eventually.